### PR TITLE
Fixes and enhancements in the local provider with extensions

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement 

**What this PR does / why we need it**:
Fixes and enhancements in the local provider with extensions

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by oliver-goetz** [comment]
Thank you 👍 

Please add a check for `project.yaml` [here](https://github.com/gardener/gardener/blob/b9ecd0bd6699be684c0570b1dd28807489e6363b/example/provider-extensions/seed/configure-seed.sh#L102-L106), that the setup fails early in case someone did not create the file.
```
ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/project/project.yaml
```

**Comment by vpnachev** [comment]
Thanks for the suggestion!
I just changed it to 
```shell
ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/project/project.yaml
```

as it looks the correct check to me.

File: example/provider-extensions/seed/configure-seed.sh [hunk 1]
```diff
@@ -104,6 +104,7 @@ ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/controlpl
 ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/project/credentials/infrastructure-secrets.yaml
 ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/project/credentials/secretbindings.yaml
 ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/"$gardenlet_values" "$REPO_ROOT_DIR"/example/provider-extensions/gardenlet/values.yaml.tmpl
+ensure-config-file "$REPO_ROOT_DIR"/example/provider-extensions/garden/project/project.yaml
```
Comment (oliver-goetz) [inline example/provider-extensions/seed/configure-seed.sh:107]: There is downside with the new templates and the `ensure-config-file` function. `infrastructure-secrets.yaml.tmpl`, `secretbindings.yaml.tmpl` and `project.yaml.tmpl` contain invalid data.
Thus, when you start from a fresh installation, those files are copied without any error. Then, the preparation of the seed cluster starts, skaffold builds the Gardener controlplane components and finally there will be an error when skaffold deploys the yaml files.
I would prefer to check for the existence of the yaml files with `check-not-initial` here. This leaves it to the user to copy those files and enter the missing values. In this case `make gardener-extensions-up` fails immediately and not after a couple of minutes.
Alternatively, you you `ensure-config-file` first and then `check-not-initial` for some attributes of these files similar to how it is done for `domain-secrets.yaml`.

Comment (vpnachev) [reply example/provider-extensions/seed/configure-seed.sh:107]: I think I get what you are talking about, especially after I had several consecutive failures of `make gardener-extensions-up` the first time I tried it . All the time it was complaining about some broken file that was not reported in the previous attempt.  

I will take a look on this.

Comment (vpnachev) [reply example/provider-extensions/seed/configure-seed.sh:107]: I pushed some changes with 4a3610f5cb284bc01cab9a1081da359551acd705 that should address your concerns, can you please take a look again?

Comment (oliver-goetz) [reply example/provider-extensions/seed/configure-seed.sh:107]: yes, looks good now. Thanks 😄

**Pull request merged**